Main-Server Ontology Server Issues

Broken Pipe ----------------------
Hoping tomcat upgrade was not applied to karenin (as this would be an easy answer) - I've updated libraries manually, waiting for Graham to return to find out but appears to not be causing any issues ATM.

java.net.SocketException: Broken pipe
	at java.net.SocketOutputStream.socketWrite0(Native Method)
	at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:113)
	at java.net.SocketOutputStream.write(SocketOutputStream.java:159)
	at java.io.ObjectOutputStream$BlockDataOutputStream.drain(ObjectOutputStream.java:1876)
	at 
java.io.ObjectOutputStream$BlockDataOutputStream.setBlockDataMode(ObjectOutputStream.java:1785)
	at java.io.ObjectOutputStream.writeNonProxyDesc(ObjectOutputStream.java:1285)
	at java.io.ObjectOutputStream.writeClassDesc(ObjectOutputStream.java:1230)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1426)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1177)
	at java.io.ObjectOutputStream.writeFatalException(ObjectOutputStream.java:1576)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:350)
	at uk.ac.ed.vfb.dao.client_server.Server$ClientHandler.sendObjectToClient(Server.java:124)
	at uk.ac.ed.vfb.dao.client_server.Server$ClientHandler.run(Server.java:153)


Brain syntax error ------------------
This is odd as not occurring on dev servers with same owl and same library files and just for a few individuals? (VFB_00015114) which displays fine. http://www.virtualflybrain.org/owl/VFB_00015114  

uk.ac.ebi.brain.error.ClassExpressionException: org.semanticweb.owlapi.expression.ParserException: Encountered VFB_00015114 at line 1 column 1. Expected one of:
	Class name
	Object property name
	Data property name
	not
	inverse
	(
	{

	at uk.ac.ebi.brain.core.Brain.parseClassExpression(Brain.java:1215)
	at uk.ac.ebi.brain.core.Brain.getSubClasses(Brain.java:1516)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryEngineBrain.askQuery(DLQueryEngineBrain.java:44)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryServer.askQuery(DLQueryServer.java:94)
	at uk.ac.ed.vfb.dao.client_server.Server$ClientHandler.run(Server.java:151)
Caused by: org.semanticweb.owlapi.expression.ParserException: Encountered VFB_00015114 at line 1 column 1. Expected one of:
	Class name
	Object property name
	Data property name
	not
	inverse
	(
	{

	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.createException(ManchesterOWLSyntaxEditorParser.java:2772)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseNonNaryClassExpression(ManchesterOWLSyntaxEditorParser.java:724)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseUnion(ManchesterOWLSyntaxEditorParser.java:625)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseIntersection(ManchesterOWLSyntaxEditorParser.java:603)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseClassExpression(ManchesterOWLSyntaxEditorParser.java:591)
	at uk.ac.ebi.brain.core.Brain.parseClassExpression(Brain.java:1213)
	... 4 more
java.lang.NullPointerException
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryEngineBrain.askQuery(DLQueryEngineBrain.java:50)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryServer.askQuery(DLQueryServer.java:94)
	at uk.ac.ed.vfb.dao.client_server.Server$ClientHandler.run(Server.java:151)



Brain syntax error ------------------
This is odd as not occurring on dev servers with same owl and same library files and just for a few individuals? (VFB_00022541) which displays fine. http://www.virtualflybrain.org/owl/VFB_00022541  


uk.ac.ebi.brain.error.ClassExpressionException: org.semanticweb.owlapi.expression.ParserException: Encountered VFB_00022541 at line 1 column 1. Expected one of:
	Class name
	Object property name
	Data property name
	not
	inverse
	(
	{

	at uk.ac.ebi.brain.core.Brain.parseClassExpression(Brain.java:1215)
	at uk.ac.ebi.brain.core.Brain.getSubClasses(Brain.java:1516)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryEngineBrain.askQuery(DLQueryEngineBrain.java:44)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryServer.askQuery(DLQueryServer.java:94)
	at uk.ac.ed.vfb.dao.client_server.Server$ClientHandler.run(Server.java:151)
Caused by: org.semanticweb.owlapi.expression.ParserException: Encountered VFB_00022541 at line 1 column 1. Expected one of:
	Class name
	Object property name
	Data property name
	not
	inverse
	(
	{

	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.createException(ManchesterOWLSyntaxEditorParser.java:2772)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseNonNaryClassExpression(ManchesterOWLSyntaxEditorParser.java:724)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseUnion(ManchesterOWLSyntaxEditorParser.java:625)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseIntersection(ManchesterOWLSyntaxEditorParser.java:603)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseClassExpression(ManchesterOWLSyntaxEditorParser.java:591)
	at uk.ac.ebi.brain.core.Brain.parseClassExpression(Brain.java:1213)
	... 4 more
java.lang.NullPointerException
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryEngineBrain.askQuery(DLQueryEngineBrain.java:50)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryServer.askQuery(DLQueryServer.java:94)
	at uk.ac.ed.vfb.dao.client_server.Server$ClientHandler.run(Server.java:151)
uk.ac.ebi.brain.error.ClassExpressionException: org.semanticweb.owlapi.expression.ParserException: Encountered VFB_00012154 at line 1 column 18. Expected one of:
	Class name
	Self
	(
	{

	at uk.ac.ebi.brain.core.Brain.parseClassExpression(Brain.java:1215)
	at uk.ac.ebi.brain.core.Brain.getSubClasses(Brain.java:1516)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryEngineBrain.askQuery(DLQueryEngineBrain.java:44)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryServer.askQuery(DLQueryServer.java:94)
	at uk.ac.ed.vfb.dao.client_server.Server$ClientHandler.run(Server.java:151)
Caused by: org.semanticweb.owlapi.expression.ParserException: Encountered VFB_00012154 at line 1 column 18. Expected one of:
	Class name
	Self
	(
	{

	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.createException(ManchesterOWLSyntaxEditorParser.java:2772)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseObjectRestriction(ManchesterOWLSyntaxEditorParser.java:745)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseNonNaryClassExpression(ManchesterOWLSyntaxEditorParser.java:709)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseUnion(ManchesterOWLSyntaxEditorParser.java:625)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseIntersection(ManchesterOWLSyntaxEditorParser.java:603)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseClassExpression(ManchesterOWLSyntaxEditorParser.java:591)
	at uk.ac.ebi.brain.core.Brain.parseClassExpression(Brain.java:1213)
	... 4 more
java.lang.NullPointerException
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryEngineBrain.askQuery(DLQueryEngineBrain.java:50)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryServer.askQuery(DLQueryServer.java:94)
	at uk.ac.ed.vfb.dao.client_server.Server$ClientHandler.run(Server.java:151)

Added extra delays before restart of tomcat after ontology server restart  - resolved Broken Pipe.

Still getting:

2015-02-27 11:06:31,090 (www.virtualflybrain.org) INFO [uk.ac.ed.vfb.dao.client_server.Server:139] Query: subclass&VFB_00004150
uk.ac.ebi.brain.error.ClassExpressionException: org.semanticweb.owlapi.expression.ParserException: Encountered VFB_00004150 at line 1 column 1. Expected one of:
	Class name
	Object property name
	Data property name
	not
	inverse
	(
	{

	at uk.ac.ebi.brain.core.Brain.parseClassExpression(Brain.java:1215)
	at uk.ac.ebi.brain.core.Brain.getSubClasses(Brain.java:1516)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryEngineBrain.askQuery(DLQueryEngineBrain.java:44)
	at uk.ac.ed.vfb.dao.client_server.server_includes.DLQueryServer.askQuery(DLQueryServer.java:94)
	at uk.ac.ed.vfb.dao.client_server.Server$ClientHandler.run(Server.java:153)
Caused by: org.semanticweb.owlapi.expression.ParserException: Encountered VFB_00004150 at line 1 column 1. Expected one of:
	Class name
	Object property name
	Data property name
	not
	inverse
	(
	{

	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.createException(ManchesterOWLSyntaxEditorParser.java:2772)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseNonNaryClassExpression(ManchesterOWLSyntaxEditorParser.java:724)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseUnion(ManchesterOWLSyntaxEditorParser.java:625)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseIntersection(ManchesterOWLSyntaxEditorParser.java:603)
	at org.coode.owlapi.manchesterowlsyntax.ManchesterOWLSyntaxEditorParser.parseClassExpression(ManchesterOWLSyntaxEditorParser.java:591)
	at uk.ac.ebi.brain.core.Brain.parseClassExpression(Brain.java:1213)
	... 4 more
2015-02-27 11:06:31,091 (www.virtualflybrain.org) ERROR [uk.ac.ed.vfb.dao.client_server.Server:163] Ontology server NullPointerException:
java.lang.NullPointerException

--- a/src/uk/ac/ed/vfb/web/OntBeanController.java
+++ b/src/uk/ac/ed/vfb/web/OntBeanController.java
@@ -37,11 +37,15 @@ public class OntBeanController implements Controller {
 				return null;
 			}else{
 				id = OntBean.idAsOWL(req.getParameter("id"));
-				ob = (OntBeanIndividual)this.obm.getBeanForId(id);
-				modelAndView.addObject("beanType", "ind");
+				
 			}
 		}else{
 			id = OntBean.idAsOBO(req.getParameter("fbId"));
+		}
+		if (id.contains("VFB")){
+			ob = (OntBeanIndividual)this.obm.getBeanForId(id);
+			modelAndView.addObject("beanType", "ind");
+		}else{
 			ob = this.obm.getBeanForId(id);
 			modelAndView.addObject("beanType", "ont");
 		}

